packages/common/src/i18n/format_date.ts

```diff
@@ -360,17 +360,27 @@ function timeZoneGetter(width: ZoneWidth): DateFormatter {
 }
 
 const JANUARY = 0;
+const SUNDAY = 0;
 const THURSDAY = 4;
+
 function getFirstThursdayOfYear(year: number) {
   const firstDayOfYear = (new Date(year, JANUARY, 1)).getDay();
   return new Date(
       year, 0, 1 + ((firstDayOfYear <= THURSDAY) ? THURSDAY : THURSDAY + 7) - firstDayOfYear);
 }
 
-function getThursdayThisWeek(datetime: Date) {
-  return new Date(
-      datetime.getFullYear(), datetime.getMonth(),
-      datetime.getDate() + (THURSDAY - datetime.getDay()));
+/**
+ * Gets the date of Thursday this week. By default, it considers a week to be from Sunday to
+ * Saturday. If called with useISOFormat as true, it considers a week to be from Monday to Sunday,
+ * as defined in the ISO 8601 format.
+ */
+function getThursdayThisWeek(datetime: Date, useISOFormat = false): Date {
```

Comment:
Since it is not that clear to me, at a glance, what the `useISOFormat` does, I would consider renaming it... WDYT @AndrewKushnir ?

```suggestion
function getThursdayThisWeek(datetime: Date, weekStartMonday = false): Date {
```

or 

```suggestion
function getThursdayThisWeek(datetime: Date, useISOWeekStart = false): Date {
```

Reply:
Updated.